src/GitHub.App/Services/ModelService.cs

```diff
@@ -134,6 +135,35 @@ public IObservable<AccountCacheItem> GetUserFromCache()
             return Observable.Defer(() => hostCache.GetObject<AccountCacheItem>("user"));
         }
 
+        public ITrackingCollection<IPullRequestModel> GetPullRequests(ISimpleRepositoryModel repo)
+        {
+            // Since the api to list pull requests returns all the data for each pr, cache each pr in its own entry
+            // and also cache an index that contains all the keys for each pr. This way we can fetch prs in bulk
+            // but also individually without duplicating information. We store things in a custom observable collection
+            // that checks whether an item is being updated (coming from the live stream after being retrieved from cache)
+            // and replaces it instead of appending, so items get refreshed in-place as they come in.
+
+            var keyobs = GetUserFromCache()
+                .Select(user => string.Format(CultureInfo.InvariantCulture, "{0}|{1}|pr", user.Login, repo.Name));
+
+            TrackingCollection<IPullRequestModel> col = null;
+
+            var source = Observable.Defer(() => keyobs
+                .SelectMany(key =>
+                    hostCache.GetAndFetchLatestFromIndex(key, () =>
+                        apiClient.GetPullRequestsForRepository(repo.CloneUrl.Owner, repo.CloneUrl.RepositoryName)
+                                 .Select(PullRequestCacheItem.Create),
+                        item => col.RemoveItem(Create(item)),
```

Comment:
I fixed the calling code, it was needlessly convoluted as you noted.

The constructor is handy for the tests right now, and will also be handy in other use cases where there's no need for an explicit `ctor` + `Listen` call (it's just required here because the observable itself references the collection, but that might not necessarily be true in other situations).